### ISSUE TITLE
Test dataset using the command line

### ISSUE THREAD

**Issue by luistelmocosta** [dotnet/machinelearning#5070]
Hello, I am using MLNET using the CLI and I have a doubt regarding the test dataset. 

`mlnet auto-train --task multiclass-classification --dataset "xxxx.tsv" --ignore-columns "Id,Task,CreatedOn" --test-dataset "xxx.tsv" --max-exploration-time 600 --label-column-name Type`

The training results look very optimistic and I was wondering if it is not tuning on the test dataset which would not be ideal. Is the test dataset being used only for "inference" after the training is complete? Do I need to pass an evaluation dataset also so the fine-tuning will be made on that or is none is given it is doing Cross Validation?

Kind regards

**Comment by Lynx1820** [comment]
Hi @luistelmocosta 

The tuning is only done on the training set. @justinormont maybe you can comment more on CV?

**Comment by justinormont** [comment]
@luistelmocosta: 

> Do I need to pass an evaluation dataset also so the fine-tuning will be made on that or is none is given it is doing Cross Validation?

The values reported during sweeping are the validation metrics. 

Since you are not providing a validation dataset, one is split from your training dataset. This is done in either a cross-validation mode (if your [dataset is <15K rows](https://github.com/dotnet/machinelearning/blob/54b2b041da42b18f947b5833e64c9763108c7d32/src/Microsoft.ML.AutoML/API/ExperimentBase.cs#L99-L116)), or train/validate mode (if your dataset is larger).

If your dataset rows are not IID, the AutoML may do better if provided with a pre-created validation dataset. The most common reason for this is when your dataset is time-dependent. If there is a time dependence in your dataset, you would want the oldest data in the training dataset, newer data in the validation dataset, and newest data in the test dataset. This helps the metrics be more representative of how the model will perform in production; it can also assist in the sweeping process for similar reasons. More info: [leakage](https://en.wikipedia.org/wiki/Leakage_(machine_learning)).

> Is the test dataset being used only for "inference" after the training is complete?

The test dataset is fully held out of the sweeping process and reserved for a final test set metrics, which is produced only by the CodeGen'd C# project. The test set metrics are the final estimate of how well your model will do once launched in production.

Work item for AutoML:
Calculate and print the test set metrics once the one final model is chosen. This could either be done in the CLI or Model Builder without modifying the current AutoML APIs, or by adding an AutoML API which takes in the test dataset and returns the test set metrics along with the current sweep results. /cc @LittleLittleCloud, @JakeRadMSFT

**Comment by luistelmocosta** [comment]
@justinormont thank you very much for your reply, I think I got it. 
Also, your suggestion makes perfect sense. At the end of the training process, I would be more interested in seeing the results on my test dataset, otherwise and as I mentioned above, the results will always look optimistic which might induce some error.
